utils/src/main/scala/za/co/absa/enceladus/utils/udf/UDFLibrary.scala

```diff
@@ -64,6 +69,20 @@ case class UDFLibrary()(implicit val spark: SparkSession) {
   spark.udf.register(errorColumnAppend,
                      UDFLibrary.errorColumnAppend,
                      ArrayType.apply(ErrorMessage.errorColSchema, containsNull = false))
+
+
+  spark.udf.register(uuid, { () => UUID.randomUUID().toString })
+
+  private val pseudoRandom = new Random(seed = 22L) // scalastyle:ignore magic.number
```

Comment:
The instance of `Random` might be created on each executor separately. Meaning that if more than 1 executor is used, it will repeat keys for each executor.

This is theory. But I'm wondering if pseudo UUID is going to create duplicate keys when more than 1 executor is used?

Reply:
Sadly, spot on. The current implementation indeed only works as expected with only one executor, otherwise, the ids get duplicate :(

Reply:
Great catch

Reply:
Spark supports a monotonically increasing numbers using this function: `monotonicallyIncreasingId`. The pseudo-UUID generation UDF can take a `long` s a parameter and add it to the generated numbers.

Still, the results will depend on the number of executors. But there shouldn't be duplicates.

Let's not rush with this to do it properly.
@HuvarVer , @Zejnilovic , what are your ideal requirements for testability?

Reply:
Well if we believe in such a "simple function" as adding UUIDs to the data, from testing perspective we would probably be ok with just `uuid` and `no` options. I would like to hear @HuvarVer ideas

Reply:
I think, from the testing perspective, that we should be ok with checking that there is the column with id in the `UUID` case and that there is no column like that in the `no` case. I think there is no reason for checking the value of nonstatic value.

Reply:
Then maybe leave it as is. If we need more sophisticated pseudo UUID generation, we can always improve it later.

Reply:
I have actually found another way to create the pseudoID values -- for the very least, it is oblivious to the division of work by executors: I calculate a hash all the columns and use this as a byte-array basis to feed the UUID creation. That way, it is completely stable no matter the way you run it.

Of course, it does directly guarantee uniqueness (the seeded-random version did not either), but I consider it to be an improvement, so it seems to me to be worth replacing what I had before. In a practical sense, if the whole rows are unique, these pseudoIDs will not collide much. From testing PoV, it could work. @Zejnilovic, @HuvarVer?

Leveraging the `monotonicallyIncreasingId` could be beneficial if we wanted to prioritize uniqueness over stability. I can implement this as well (as another mode) if it is desired but I believe quite strongly that the former hash-based solution complies better to the originally intended pseudo mode (i.e. prioritize stability over uniqueness)

Your thoughts?

Reply:
That sounds really good. The hash solution 👍 
The `monotonicallyIncreasingId` would be good too. Decide if you add it to this PR or as a separate feature. (I vote for inclusion)

Reply:
From discussion with @dk1844 a small drawback of the hash was realized - for the same row repeated in the dataframe it creates the same value, so it stops to be key per-se. But we believe for the purposes it's still good enough.
Please first of all testers confirm (@Zejnilovic @HuvarVer ), but anybody is welcomed to comment (@yruslan , @AdrianOlosutean , ... )

Reply:
So making your own key using hashed rows is actually something that Hermes does when people don't supply their keys. So I like it.

The duplicities are something to look out for. I have an idea for Hermes that might be applied here, but it is not that performant. Generate your hashes, then check for duplicate rows and append something to them if there are duplicitous.

I know this idea is broad but might spark something.

`monotonicallyIncreasingId` are not needed then.

Reply:
> Generate your hashes, then check for duplicate rows and append something to them if there are duplicitous.

I briefly considered this as well, but I did not seem worthwhile. In my limited experience, the rows seem to be unique in practice.

Reply:
The beauty of the current implementation is, that we can add strategies if their need emerges. What @Zejnilovic  wrote is one possible solution for a more robust unique id. And someones till might want the `monotonicallyIncreasingId` as well.
😀

Reply:
If all columns of 2 rows are the same, we can consider them the same, and they are ok to have the same `enceladus_row_id`, especially for `stable uuid`.

Reply:
If we have hash solution, monotonic id is not needed. It has its caveats, let's not do this for now.